src/SharedUtilities/Domains.jl

```diff
@@ -401,22 +401,121 @@ abstract type AbstractVegetationDomain{FT} <: AbstractDomain{FT} end
 Domain for a single bulk plant with roots of varying depths. The user needs
 to specify the depths of the root tips as wel as the heights of the
 compartments to be modeled within the plant. The compartment heights
-are expected to be sorted in ascending order.
+are expected to be sorted in ascending order. The number of stem and leaf
+compartments the plant should be divided into must be specified as well.
 """
 struct PlantHydraulicsDomain{FT} <: AbstractVegetationDomain{FT}
     "The depth of the root tips, in meters"
     root_depths::Vector{FT}
+    "The number of stem compartments for the plant"
+    n_stem::Int64
+    "The number of leaf compartments for the plant"
+    n_leaf::Int64
+    "The height of the center of each leaf compartment/stem compartment, in meters"
+    compartment_midpoints::Vector{FT}
     "The height of the stem and leaf compartments, in meters"
```

Comment:
can we be a little clearer in the doc string - "the height of the compartment top face, in meters" or something?